# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLSampler.h
# Commit: ccfa767dba64
# Full Hash: ccfa767dba644dc193e0246eb8e8ff3377e8b8a5
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Client-side bindings mirror for precise CC, and merge similar codepaths. r=handyman
#   
#   * Context loss using RAII
#   * Move Program reflection Client-side
#   
# ==============================================================================

diff -r 71c122ac0ca7 -r ccfa767dba64 dom/canvas/WebGLSampler.h
--- a/dom/canvas/WebGLSampler.h	Wed Jan 08 22:19:14 2020 +0000
+++ b/dom/canvas/WebGLSampler.h	Wed Jan 08 22:19:16 2020 +0000
@@ -6,18 +6,15 @@
 #ifndef WEBGL_SAMPLER_H_
 #define WEBGL_SAMPLER_H_
 
-#include "mozilla/LinkedList.h"
-#include "nsWrapperCache.h"
 #include "WebGLObjectModel.h"
 #include "WebGLStrongTypes.h"
 #include "WebGLTexture.h"
 
 namespace mozilla {
 
-class WebGLSampler final : public WebGLRefCountedObject<WebGLSampler>,
-                           public LinkedListElement<WebGLSampler>,
+class WebGLSampler final : public WebGLContextBoundObject,
                            public CacheInvalidator {
-  NS_INLINE_DECL_REFCOUNTING(WebGLSampler)
+  MOZ_DECLARE_REFCOUNTED_VIRTUAL_TYPENAME(WebGLSampler, override)
 
  public:
   const GLuint mGLName;
@@ -32,10 +29,7 @@
   ~WebGLSampler();
 
  public:
-  void Delete();
-
   void SamplerParameter(GLenum pname, const FloatOrInt& param);
-
   const auto& State() const { return mState; }
 };
 